app/frontend/src/features/communities/events/EventForm.tsx

```diff
@@ -0,0 +1,257 @@
+import { UseMutateFunction } from "react-query";
+import { Dayjs } from "utils/dayjs";
+import type { GeocodeResult } from "utils/hooks";
+import makeStyles from "utils/makeStyles";
+
+import {
+  END_DATE,
+  END_TIME,
+  EVENT_DETAILS,
+  EVENT_IMAGE_INPUT_ALT,
+  LINK_REQUIRED,
+  LOCATION,
+  LOCATION_REQUIRED,
+  START_DATE,
+  START_TIME,
+  TITLE_REQUIRED,
+  VIRTUAL_EVENT,
+  VIRTUAL_EVENT_LINK,
+} from "./constants";
+
+const useStyles = makeStyles((theme) => ({
+  imageInput: {
+    marginBlockStart: theme.spacing(3),
+  },
+  form: {
+    display: "grid",
+    rowGap: theme.spacing(3),
+    marginBlockEnd: theme.spacing(3),
+  },
+  duoContainer: {
+    display: "grid",
+    gridTemplateColumns: "1fr",
+    gap: theme.spacing(3, 2),
+    [theme.breakpoints.up("md")]: {
+      gridTemplateColumns: "1fr 1fr",
+    },
+  },
+  locationContainer: {
+    minHeight: theme.typography.pxToRem(66),
+  },
+  endDateTimeButton: {
+    justifySelf: "start",
+  },
+  isOnlineCheckbox: {
+    width: "50%",
+  },
+  eventDetailsContainer: {
+    display: "grid",
+    rowGap: theme.spacing(1),
+  },
+  createEventButton: {
+    justifySelf: "end",
+  },
+}));
+
+interface BaseEventData {
+  content: string;
+  title: string;
+  startDate: Dayjs;
+  endDate: Dayjs;
+  startTime: Dayjs;
+  endTime: Dayjs;
+  isOnline: boolean;
+  eventImage?: string;
+  parentCommunityId?: number;
+  link?: string;
+  location?: GeocodeResult;
+}
+interface OfflineEventData extends BaseEventData {
+  isOnline: false;
+  location: GeocodeResult;
+}
+
+interface OnlineEventData extends BaseEventData {
+  isOnline: true;
+  link: string;
+  parentCommunityId: number;
+}
+
+export type CreateEventData = OfflineEventData | OnlineEventData;
+
+interface EventFormProps {
+  error: GrpcError | null;
+  mutate: UseMutateFunction<unknown, GrpcError, CreateEventData, unknown>;
+  isMutationLoading: boolean;
+  isMutationSuccess: boolean;
+  successMessage: string;
+  title: string;
+}
+
+export default function EventForm({
+  error,
+  mutate,
+  isMutationLoading,
+  isMutationSuccess,
+  successMessage,
+  title,
+}: EventFormProps) {
+  const classes = useStyles();
+  const { control, errors, handleSubmit, register, watch } =
+    useForm<CreateEventData>();
+
+  const isOnline = watch("isOnline", false);
+
+  const onSubmit = handleSubmit(
+    (data) => {
+      mutate(data);
+    },
+    (errors) => {
+      if (errors.eventImage) {
+        window.scroll({ top: 0, behavior: "smooth" });
+      }
+    }
+  );
+
+  return (
+    <>
+      <ImageInput
+        alt={EVENT_IMAGE_INPUT_ALT}
+        className={classes.imageInput}
+        control={control}
+        id="event-image-input"
+        name="eventImage"
+        type="rect"
+      />
+      <PageTitle>{title}</PageTitle>
+      {error || errors.eventImage || errors.location ? (
+        <Alert severity="error">
+          {error?.message ||
+            errors.eventImage?.message ||
+            // @ts-expect-error
```

Comment:
add a reason - "react-hook-form typing incorrect for nested types"

Reply:
I'll just write for the comment for this first `ts-expect-error`. The rest follows the same pattern so I don't think the comment needs to be repeated all over the place